test/jdk/sun/tools/jcmd/TestJcmdNoAgentLoad.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import static jdk.test.lib.Asserts.*;
+
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import com.sun.management.HotSpotDiagnosticMXBean;
+import com.sun.management.VMOption;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.Utils;
+
+/*
+ * @test id=default
+ * @bug 8304438
+ * @summary JVMTI.agent_load should obey EnableDynamicAgentLoading (by default)
+ * @library /test/lib
+ * @run main/othervm TestJcmdNoAgentLoad
+ */
+
+/*
+ * @test id=disabled
+ * @bug 8304438
+ * @summary JVMTI.agent_load should obey EnableDynamicAgentLoading (disabled)
+ * @library /test/lib
+ * @run main/othervm -XX:-EnableDynamicAgentLoading TestJcmdNoAgentLoad
+ */
+
+/*
+ * @test id=enabled
+ * @bug 8304438
+ * @summary JVMTI.agent_load should obey EnableDynamicAgentLoading (enabled)
+ * @library /test/lib
+ * @run main/othervm -XX:+EnableDynamicAgentLoading TestJcmdNoAgentLoad
+ */
+
+
+public class TestJcmdNoAgentLoad {
+    // The Agent.jar does not exist. Error with the PTRN message
+    // is expected before an attempt to load the Agent.jar.
+    private static final String[] CMD = new String[] { "JVMTI.agent_load", "Agent.jar" };
+    private static final String PTRN = "Dynamic agent loading is not enabled";
+    private static boolean enableDynLoad = true;
```

Comment:
It might be clear to change this to be a static final field name "dynamicLoadingEnabled", just suggesting "enabled" rather than "enable" as the usage in this test is to see if the option is enabled.

Reply:
Good suggestion, thanks. Fixed now.